File: java/p2/src/p2/types/table/EventTable.java
package p2.types.table;

import java.util.Hashtable;
import p2.types.basic.Tuple;
import p2.types.basic.TupleSet;
import p2.types.basic.TypeList;
import p2.types.exception.UpdateException;

public class EventTable extends Table {

	public EventTable(TableName name, TypeList types) {
		super(name, Type.EVENT, null, types);
	}
	
	@Override
	public TupleSet insert(TupleSet tuples, TupleSet deletions) throws UpdateException {
		return tuples;
	}
	
	@Override
	public TupleSet delete(TupleSet tuples) throws UpdateException {
		return tuples;
	}

	protected boolean insert(Tuple t) throws UpdateException {
		throw new UpdateException("Can't insert tuple in event table");
	}

	@Override
	protected boolean delete(Tuple t) throws UpdateException {
		throw new UpdateException("Can't remove tuple from event table");
	}

	@Override
	public Index primary() {
		return null;
	}

	@Override
	public Hashtable<Key, Index> secondary() {
		return null;
	}

	public TupleSet tuples() {
		return null;
	}

	@Override
	public Integer cardinality() {
		return 0;
	}

}

File: lincoln/ruby/lib/types/table/event_table.rb
require 'lib/types/table/table'

class EventTable < Table
  def initialize(name, types)
    super(name, Event, 0, 0, nil, types)
  end
  
  def insert(tuples, deletions)
    return tuples
  end
  
  def delete(tuples)
    if tuples.class != TupleSet
      raise "Can't delete a single tuple from an EventTable"
    end
    return tuples
  end
    
  def primary
    nil
  end
  
  def secondary
    nil
  end
  
  def tuples
    nil
  end
  
  def cardinality
    0
  end
end